bazel/README.md

```diff
@@ -515,6 +515,76 @@ debugging information files (`.dwp` files) and we can build a DWARF package file
 with `.dwp ` target. The `.dwp` file need to be presented in the same folder with the
 binary for a full debugging experience.
 
+# Debugging production core dumps
+
+In production, you'll likely be running a version of envoy compiled with `-c
+opt`. When we build envoy with `-c opt` it still has debug symbols attached
+which greatly bloats the binary size.
+Envoy uses fission which uses `gsplit-dwarf` to have the debug info partially in
+the `.o` but mostly in `.dwo` file for each object file.
```

Comment:
When I use clang I usually get dwp files. Are you using gcc?

Reply:
i was confused by that also - our debug images have `.dwp`

Reply:
interesting, I use clang14 to build and I just get `.dwo` in `bazel-bin` when only build envoy-static using `bazel build -c opt //source/exe:envoy-static --config=clang`

Reply:
maybe its different if you build `opt`/`dbg`

Reply:
ah we build the debug images with `-c dbg`? If so it works for debugging `-c dbg` cores but not `-c opt` cores as the code generated is different.

Reply:
it would be good if we elucidate this in these docs i think

Reply:
thinking further on this ...

we build all release assets (including for Docker dbg container) using `-c opt` - its _enforced_ with a bazel transition

this is the typical product

```console
$ tar tf ~/Downloads/release.tar.zst 
dbg/
dbg/envoy
dbg/envoy-contrib
dbg/envoy-contrib.dwp
dbg/envoy.dwp
envoy
envoy-contrib
utils/
utils/su-exec
```

so its not `opt`/`dbg` i think

Reply:
that said there are some differences i think in the 2 types of build

Reply:
so re `.dwo` and `.dwp` i think the former are compiled together to create the latter, but its the `.dwp` files that you need for debugging

Reply:
im also wondering if above sentence could be a bit more concise/user-focused - i think all we need to say is that when build with `-c opt` envoy will by default split the debug symbols out to a `.dwp` file which can be used for debugging